iocore/eventsystem/UnixEThread.cc

```diff
@@ -265,12 +304,15 @@ EThread::execute()
         while ((e = NegativeQueue.dequeue())) {
           process_event(e, EVENT_POLL);
         }
+
+        metric_wait_check = EventQueue.earliest_timeout() - cur_time;
+
```

Comment:
There is no sleep_time if NegativeQueue is not empty.
```
EventQueueExternal.dequeue_timed(cur_time, next_time, false);
```
The 3rd parameter is `false` means no condition wait and return immediately.